cmd/clusterctl/client/get_kubeconfig_test.go

```diff
@@ -48,7 +48,7 @@ func Test_clusterctlClient_GetKubeconfig(t *testing.T) {
 		{
 			name:      "returns error if unable namespace is empty",
 			client:    badClient,
-			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig)},
+			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig), UserKubeconfig: true},
 			expectErr: true,
 		},
```

Comment:
Can you also add a test case for the fallback flow?

Reply:
I wanted to, but I couldn't figure out how to setup a fake cluster. I'll give it a shot again.

Reply:
I will take a closer look at this and drop something here in the coming days.

Reply:
Thanks. There's no hurry at my end, I am traveling for the next 2 weeks, so I might not be able to work on this right away.